models/clm/CAM_bias_correction/CAM4_bias_correct.m

```diff
@@ -0,0 +1,511 @@
+clear all
+close all
+
+% This script applies a bias correction to CAM4 reanalysis meteorology roughly
+% following Yong-Fei Zhang (UT-Austin) dissertation approach. Meteorology data
+% from the site location of interest should be used as 'truth'.
+% All met variables use 'scaling' approach with exception for snow/rain precip.
+% *Warning* This script is intended to be used after running CAM4_site_grid.sh 
+
+%%%% This section requires user input %%%%%%
+% Site location. US-NR1 flux tower used as example
+ SITE_lat=40.03;
+ SITE_lon=-105.55+360; % lon should be positive (degrees East only)
+
+%% SITE_doma_area=0.00109327562271889; % m^2 cell area (unique to grid location)
+%% SITE_doma_mask=1   ; % all values =1 
+
+% Input data files
+path_CAM = '/glade/work/bmraczka/CAM4_NR1/';
+path_towermet = '/glade/work/bmraczka/SIFMIP2/tower_met_forcing/';
+ens_mem=80;  % CAM6 reanalysis provides 80 total members, 1-80 is valid
+
+% Output data files
+path_scaled_CAM = '<enter output file path here>';   
+% Include Diagnostics?  true/false
+Diagnostics=true;
+
+% End of user input section
+%-----------------------------------------------------------------------------
+
+% Site level extract CAM4:
+% One file includes all met forcing.
+% noleap calendar
+% SOLAR: a2x6h_Faxa_swvdf, a2x6h_Faxa_swndf, a2x6h_Faxa_swvdr, a2x6h_Faxa_swndr, a2x6h_Faxa_lwdn
+% PRECIP: a2x6h_Faxa_rainc, a2x6h_Faxa_rainl, a2x6h_Faxa_snowc, a2x6h_Faxa_snowl
+% MET:    a2x6h_Sa_pbot, a2x6h_Sa_shum, a2x6h_Sa_tbot, a2x6h_Sa_v, a2x6h_Sa_u
+% unchanged: a2x6h_Sa_pslv, a2x6h_Sa_dens, a2x6h_Sa_ptem, a2x6h_Sa_z
+
+
+% 'yearstr' is list of years where both CAM4 reanalysis and tower met forcing
+% is available and the years we will create the bias-correct/scaled reanalysis product
+yearstr={'2001','2002','2003','2004','2005','2006','2007','2008','2009','2010'};
+% 'yeartower' adds previous year to cover the UTC to MST shift 
+yeartower={'2000','2001','2002','2003','2004','2005','2006','2007','2008','2009','2010'};
+
+% Generate ensemble cell array
+ens_range=[1:ens_mem];
+enstr=cell(1,ens_mem);
+
+for i=1:ens_mem
+    enstr{i}=sprintf('%04d', ens_range(i));
+end
+
+% Check for existence of output file from CAM6_site_grid.sh

```

Comment:
Should probably be CAM4_site_grid.sh

Reply:
corrected: 1846d1a35de0137693b6299dfc47c694366b84f3